### ISSUE TITLE
NullPointerException for MaxLengthExactValuesInStringFieldsFuzzer with uri or uuid format

### ISSUE THREAD

**Issue by jamesfulford** [Endava/cats#10]
Stack trace for NullPointerException pointed here: https://github.com/Endava/cats/blob/master/src/main/java/com/endava/cats/fuzzer/fields/base/ExactValuesInFieldsFuzzer.java#L40

I found it only worked if I removed any mention of "format": "uri" or "format": "uuid" from the spec for my input parameters. Not sure what the problem is with those?

**Comment by en-milie** [comment]
Can you please post an example similar to your API specs? Ideally with the payload schema. It's strange that it throws a NPE if maxLength is defined....

**Comment by jamesfulford** [comment]
I ran with this subset of my spec, got issue again:
```
{
    "components": {
        "schemas": {
            "SubscriptionCreateApiResponse": {
                "properties": {
                    "data": {
                        "$ref": "#/components/schemas/SubscriptionResponse"
                    },
                    "meta": {
                        "$ref": "#/components/schemas/MetaSubInfo"
                    }
                },
                "type": "object"
            },
            "SubscriptionRequest": {
                "properties": {
                    "description": {
                        "description": "Client-defined string describing the subscription.",
                        "example": "The subscription my integration uses to synchronize with WxCC",
                        "maxLength": 1024,
                        "minLength": 0,
                        "type": "string"
                    },
                    "destinationUrl": {
                        "description": "URL to which webhooks will be posted. Must be HTTPS on an IANA-listed top-level domain name (e.g. .com) with a path (at least /). No query parameters, userinfo, non-443 ports, or fragments allowed. We do not treat this field as sensitive data, so do not use secrets in this URL such as tokens or API keys.",
                        "example": "https://www.cisco.com/",
                        "format": "uri",
                        "pattern": "^(https?)://.*$",
                        "type": "string"
                    },
                    "eventTypes": {
                        "description": "Types of events to which the subscription will listen.",
                        "example": [
                            "agent:login",
                            "agent:logout"
                        ],
                        "items": {
                            "type": "string"
                        },
                        "maxItems": 32,
                        "minItems": 1,
                        "type": "array"
                    },
                    "name": {
                        "description": "Client-defined string naming the subscription.",
                        "example": "My Integration's Subscription",
                        "maxLength": 64,
                        "minLength": 4,
                        "type": "string"
                    },
                    "orgId": {
                        "description": "Organization ID to be used for this operation. If unspecified, the Organization ID is inferred from the token. The token must have permission to interact with the organization.",
                        "example": "97cdbf45-ebe2-4687-8341-44d5c7abf101",
                        "format": "uuid",
                        "type": "string"
                    },
                    "secret": {
                        "description": "Secret string used to sign payloads sent to the destination URL.",
                        "example": "my-32-plus-character-long-secret",
                        "maxLength": 256,
                        "minLength": 32,
                        "type": "string"
                    }
                },
                "required": [
                    "destinationUrl",
                    "eventTypes",
                    "name"
                ],
                "type": "object"
            }
        },
        "securitySchemes": {
            "bearer-key": {
                "bearerFormat": "JWT",
                "scheme": "bearer",
                "type": "http"
            }
        }
    },
    "info": {
        "contact": {
            "name": "jamesfulford"
        },
        "description": "APIs for registering subscriptions to receive event notifications",
        "title": "Subscription API",
        "version": "1.0.0"
    },
    "openapi": "3.0.1",
    "paths": {
        "/v1/subscriptions": {
            "post": {
                "description": "Create a subscription which would allow consumers to listen to events. If creating a subscription causes the org-level limit to be exceeded, the subscription registration will be denied.",
                "operationId": "createSubscription",
                "parameters": [
                    {
                        "description": "Tracking ID to use for this operation, for traceability, debugging, and error reporting purposes. If not provided, we will generate one for you.",
                        "example": "NOTIFS_9bcdc696-57fa-4e91-b5aa-57a66a347c23",
                        "in": "header",
                        "name": "TrackingId",
                        "required": false,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "requestBody": {
                    "content": {
                        "application/json": {
                            "schema": {
                                "$ref": "#/components/schemas/SubscriptionRequest"
                            }
                        }
                    },
                    "required": true
                },
                "responses": {
                    "201": {
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/SubscriptionCreateApiResponse"
                                }
                            }
                        },
                        "description": "Created"
                    }
                },
                "summary": "Register Subscription",
                "tags": [
                    "Subscriptions"
                ]
            }
        }
    },
    "servers": [
        {
            "description": "Generated server url",
            "url": "http://localhost:8080"
        }
    ],
    "tags": [
        {
            "description": "The Subscription resource represents a demand to be notified of events via webhooks. Registering and managing Subscriptions requires an administrator role and the appropriate `cjp:config_read` or `cjp:config_write` scopes. For more on how to integrate Subscriptions and webhooks, consult the [Using Webhooks guide](/documentation/guides/using-webhooks).",
            "name": "Subscriptions"
        }
    ]
}
```

I run with command:
```
./cats.jar \
        --contract=../api.json \
        --headers=.headers.creds \
        --refData=.refdata.creds \
        --server=$TARGET_API_URI \
        --urlParams=orgId:$MAIN_ORG_ID \
        --timestampReports \
        --checkFields --paths=/v1/subscriptions
```

.headers.creds contains an Authorization header, refdata.creds just says any orgId input param should be $MAIN_ORG_ID. 

Stack trace:
```
[********* ][MLEVI] ℹ info     Field [destinationUrl] schema is [StringSchema] and type [string]
[********* ][MLEVI] ✖ error    Something went wrong while running CATS!
java.lang.NullPointerException: null
        at com.endava.cats.fuzzer.fields.base.ExactValuesInFieldsFuzzer.getBoundaryValue(ExactValuesInFieldsFuzzer.java:40)
        at com.endava.cats.fuzzer.fields.base.BaseBoundaryFieldFuzzer.fuzzerGeneratedBoundaryValue(BaseBoundaryFieldFuzzer.java:57)
        at com.endava.cats.fuzzer.fields.base.BaseBoundaryFieldFuzzer.getFieldFuzzingStrategy(BaseBoundaryFieldFuzzer.java:40)
        at com.endava.cats.fuzzer.fields.base.BaseFieldsFuzzer.fuzz(BaseFieldsFuzzer.java:57)
        at com.endava.cats.fuzzer.fields.base.BaseFieldsFuzzer$$FastClassBySpringCGLIB$$3055f71f.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
        at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89)
        at com.endava.cats.aop.FuzzerLogAspect.logExecutionTime(FuzzerLogAspect.java:31)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634)
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624)
        at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)
        at com.endava.cats.fuzzer.fields.MaxLengthExactValuesInStringFieldsFuzzer$$EnhancerBySpringCGLIB$$498358a8.fuzz(<generated>)
        at com.endava.cats.CatsMain.lambda$fuzzPath$14(CatsMain.java:385)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at com.endava.cats.CatsMain.fuzzPath(CatsMain.java:383)
        at com.endava.cats.CatsMain.startFuzzing(CatsMain.java:174)
        at com.endava.cats.CatsMain.doLogic(CatsMain.java:165)
        at com.endava.cats.CatsMain.run(CatsMain.java:143)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:819)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:803)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:346)
        at com.endava.cats.CatsMain.main(CatsMain.java:102)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:467)
```

Version 6.0.2

**Comment by en-milie** [comment]
Discovered the issue. Working on a fix today. Thanks for the detailed report.

**Comment by LiamRiddell** [comment]
@en-milie  - I believe the `MinLengthExactValuesInStringFieldsFuzzer` has the same issue as well. :) I had this same issue yesterday and had to disable both of the min/max length values fuzzers.

**Comment by en-milie** [comment]
Yes, it's the same root cause.

**Comment by en-milie** [comment]
This is now fixed in 6.0.4: https://github.com/Endava/cats/releases/tag/cats-6.0.4
